lib/children.c

```diff
@@ -65,6 +66,12 @@ child_manager_unregister(pid_t pid)
     }
 }
 
+void
+child_manager_register_external_sigchld_handler(void (*callback)(int))
+{
+  sigchld_callback = callback;
```

Comment:
There are three component, where the `sigaction` is called (there are other guarantee as well):
* mainloop `setup_signals`
  It is called before the configuration parse, or any pipe is created.
* python module, when the module loaded
  There cannot be race because it is called - during the configuration parse - from the same main thread as the `setup_signals`.
* java module; this is actually for certainly called from a different thread than the other two, the one that is created by the `JVM`. But the `JVM` is started only, when the java destination `init` is called. The `init` function for the pipes are called after parsing the configuration file (only after the modules are already loaded).